### PR TITLE
Improve javadocs

### PR DESCRIPTION
**Pull request by timowest**

### CODE REVIEW

File: querydsl-core/src/main/java/com/querydsl/core/JoinFlag.java [hunk 1]
```diff
@@ -26,11 +26,13 @@
  * <p>{@code JoinFlag} instances can be used in Querydsl modules which serialize queries to String form.</p>
  *
  * @author tiwe
- *
  */
 @Immutable
 public class JoinFlag implements Serializable {
 
+    /**
+     * Position
```
Comment (Shredder121) [inline querydsl-core/src/main/java/com/querydsl/core/JoinFlag.java]: Might `The different JoinFlag positions` be a better description?

File: querydsl-core/src/main/java/com/querydsl/core/QueryFlag.java [hunk 2]
```diff
@@ -29,6 +29,9 @@
 
     private static final long serialVersionUID = -7131081607441961628L;
 
+    /**
+     * Position of flag
```
Comment (Shredder121) [inline querydsl-core/src/main/java/com/querydsl/core/QueryFlag.java]: Might `The different QueryFlag positions` be a better description?

File: querydsl-core/src/main/java/com/querydsl/core/types/dsl/CaseForEqBuilder.java [hunk 3]
```diff
@@ -136,13 +136,19 @@ protected StringExpression createResult(Class<String> type, Expression<String> l
         }.when(other).then(then);
     }
 
+    /**
+     * Step class
```
Comment (johnktims) [inline querydsl-core/src/main/java/com/querydsl/core/types/dsl/CaseForEqBuilder.java]: What is a step class?

Comment (timowest) [inline querydsl-core/src/main/java/com/querydsl/core/types/dsl/CaseForEqBuilder.java]: It's an intermediate node in the fluent builder of the case expression.

Comment (timowest) [inline querydsl-core/src/main/java/com/querydsl/core/types/dsl/CaseForEqBuilder.java]: Shall I used `Intermediate step` instead?

Comment (Shredder121) [inline querydsl-core/src/main/java/com/querydsl/core/types/dsl/CaseForEqBuilder.java]: To me that sounds better.

**Comment by timowest** [comment]
I fixed now the `Step class` usage.

**Comment by timowest** [comment]
@johnktims @Shredder121 Is this ok to merge?

**Comment by johnktims** [comment]
:+1:

**Comment by Shredder121** [comment]
Yes, this is good. :+1:

**Pull request merged**